### PR TITLE
`save_email` cleanup

### PR DESCRIPTION
**Pull request by raxyte**

### CODE REVIEW

**Review by tobim — COMMENTED** [review]
Code-wise this looks all correct.
Testing now.

File: libtenzir/builtins/connectors/email.cpp [hunk 1]
```diff
@@ -68,19 +69,30 @@ class saver final : public plugin_saver {
...
-    if (code != curl::easy::code::ok) {
-      auto err = to_error(code);
-      diagnostic::error("failed to set SMTP server request")
-        .note("server: {}", args_.endpoint)
+    if (auto err = tx.prepare(std::move(args_.endpoint))) {
+      diagnostic::error("failed to prepare SMTP server request")
+        .note("{}", err)
+        .emit(ctrl.diagnostics());
+      return err;
+    }
```
Comment (tobim) [inline libtenzir/builtins/connectors/email.cpp:77]: The `prepare` call must come before any other option setting calls to `tx`.
Move it above the `CURLOPT_UPLOAD` setting to restore compatibility with servers that don't implement VRFY.

**Review by tobim — APPROVED** [review]
I tested with the requested change applied and it worked as expected.

File: libtenzir/builtins/connectors/email.cpp (hunk 1 above)
Comment (raxyte) [reply libtenzir/builtins/connectors/email.cpp:74]: Moved up.

**Pull request merged**